drivers/serial/uart_gecko.c

```diff
@@ -13,23 +14,74 @@
 
 #define USART_PREFIX cmuClock_USART
 #define UART_PREFIX cmuClock_UART
-#define CLOCK_ID_PRFX2(prefix, suffix) prefix##suffix
-#define CLOCK_ID_PRFX(prefix, suffix) CLOCK_ID_PRFX2(prefix, suffix)
-#define CLOCK_USART(id) CLOCK_ID_PRFX(USART_PREFIX, id)
-#define CLOCK_UART(id) CLOCK_ID_PRFX(UART_PREFIX, id)
+#define CLOCK_USART(id) _CONCAT(USART_PREFIX, id)
+#define CLOCK_UART(id) _CONCAT(UART_PREFIX, id)
+
+/* Helper define to determine if SOC supports hardware flow control */
+#if ((_SILICON_LABS_32B_SERIES > 0) || \
+	(defined(_USART_ROUTEPEN_RTSPEN_MASK) && \
+	defined(_USART_ROUTEPEN_CTSPEN_MASK)))
+#define HW_FLOWCONTROL_IS_SUPPORTED_BY_SOC
+#endif
+
+#define HAS_HFC_OR(inst) DT_INST_NODE_HAS_PROP(inst, hw_flow_control) ||
```

Comment:
I missed that this was a boolean property when I first responded. Boolean properties should be tested with DT_INST_PROP() or DT_PROP(), not ...NODE_HAS_PROP(). The bindings type system knows that a boolean property is true if present and false if absent.
https://docs.zephyrproject.org/latest/guides/dts/api-usage.html#property-access

Reply:
done